v0/destinations/tiktok_ads/transform.js

```diff
@@ -33,18 +35,111 @@ function checkIfValidPhoneNumber(str) {
 }
 
+const getTrackResponse = (message, Config, event) => {
+  const pixel_code = Config.pixelCode;
+  let payload = constructPayload(message, trackMapping);
+
+  const externalId = getDestinationExternalID(message, "tiktokExternalId");
+  if (isDefinedAndNotNullAndNotEmpty(externalId)) {
+    set(payload, "context.user.external_id", externalId);
+  }
+
+  const traits = getFieldValueFromMessage(message, "traits");
+
+  // taking user properties like email and phone from traits
+  let email = get(payload, "context.user.email");
+  if (!isDefinedAndNotNullAndNotEmpty(email) && traits?.email) {
+    set(payload, "context.user.email", traits.email);
+  }
+
+  let phone_number = get(payload, "context.user.phone_number");
+  if (!isDefinedAndNotNullAndNotEmpty(phone_number) && traits?.phone) {
+    set(payload, "context.user.phone_number", traits.phone);
+  }
+
+  payload = { pixel_code, event, ...payload };
+
+  /*
+   * Hashing user related detail i.e external_id, email, phone_number
+   */
+
+  if (Config.hashUserProperties) {
+    const external_id = get(payload, "context.user.external_id");
+    if (isDefinedAndNotNullAndNotEmpty(external_id)) {
+      payload.context.user.external_id = SHA256(external_id.trim()).toString();
+    }
+
+    email = get(payload, "context.user.email");
+    if (isDefinedAndNotNullAndNotEmpty(email)) {
+      payload.context.user.email = SHA256(
+        email.trim().toLowerCase()
+      ).toString();
+    }
+
+    phone_number = get(payload, "context.user.phone_number");
+    if (isDefinedAndNotNullAndNotEmpty(phone_number)) {
+      if (checkIfValidPhoneNumber(phone_number.trim())) {
+        payload.context.user.phone_number = SHA256(
+          phone_number.trim()
+        ).toString();
+      } else {
+        throw new CustomError(
+          "Invalid phone number. Include proper country code except +86. Aborting ",
+          400
+        );
+      }
+    }
+  }
+  const response = defaultRequestConfig();
+  response.headers = {
+    "Access-Token": Config.accessToken,
+    "Content-Type": "application/json"
+  };
+
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.endpoint = TRACK_ENDPOINT;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+
+  // payload = removeUndefinedAndNullValues(payload);
+  return response;
+};
+
 const trackResponseBuilder = async (message, { Config }) => {
   const pixel_code = Config.pixelCode;
+  const eventsToStandard = Config.eventsToStandard;
 
   let event = get(message, "event");
   event = event ? event.trim().toLowerCase() : event;
   if (!event) {
     throw new CustomError("Event name is required", 400);
   }
 
-  if (eventNameMapping[event] === undefined) {
+  const standardEventsMap = getHashFromArrayWithDuplicate(eventsToStandard);
+  const trimmedEvent = event.toLowerCase().trim();
+
+  if (
+    eventNameMapping[event] === undefined &&
+    standardEventsMap &&
+    !standardEventsMap[trimmedEvent]
+  ) {
```

Comment:
```suggestion
  if (
    eventNameMapping[event] === undefined && !standardEventsMap[trimmedEvent]
  ) {
```
`getHashFromArrayWithDuplicate` function always returns one object so no need to check that.